File: sub-web-service/src/main/java/org/github/web/module/sys/menu/service/impl/PaymentServiceImpl.java
package org.github.web.module.sys.menu.service.impl;

import lombok.*;
import org.github.exception.ParamsErrorException;
import org.github.mysql.web.base.entity.PaymentMbpEntity;
import org.github.mysql.web.base.service.IPaymentMbpService;
import org.github.web.module.sys.menu.service.IPaymentService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import static java.lang.String.*;

@Service
public class PaymentServiceImpl implements IPaymentService {
  @Autowired
  private IPaymentMbpService paymentMbpService;

  @Override
  public boolean create(@NonNull PaymentMbpEntity bo) {
    return paymentMbpService.save(bo);
  }

  @Override
  public @NonNull PaymentMbpEntity get(long id) {
    val entity = paymentMbpService.getById(id);
    if (entity == null) throw new ParamsErrorException(format("[id: %s]对应的记录不存在", id));
    return entity;
  }
}

File: sub-mysql-mydb-service/src/main/java/org/github/mysql/mydb/base/service/impl/PaymentServiceImpl.kt
@file:Suppress("RedundantSemicolon")

package org.github.mysql.mydb.base.service.impl;

import org.github.mysql.mydb.base.entity.PaymentEntity;
import org.github.mysql.mydb.base.mapper.IPaymentMapper;
import org.github.mysql.mydb.base.service.IPaymentService;
import org.github.base.ServiceImpl;
import org.springframework.stereotype.Service;

/**
 * 支付表 服务实现类
 *
 * @author JYD_XL
 */
@Service
open class PaymentServiceImpl : ServiceImpl<IPaymentMapper, PaymentEntity>(), IPaymentService
